release/air_tests/air_benchmarks/workloads/torch_benchmark.py

```diff
@@ -0,0 +1,298 @@
+        node_to_rank_str,
+        "--num-workers",
+        str(num_workers),
+        "--master-addr",
+        master_addr,
+        "--master-port",
+        str(master_port),
+    ]
+
+    if use_gpu:
+        cmd += ["--use-gpu"]
+
+    run_command_on_all_nodes(cmd)
+
+
+@click.command()
+@click.option("--num-workers", type=int, default=4)
+@click.option("--num-epochs", type=int, default=4)
+@click.option("--use-gpu", is_flag=True, default=False)
+@click.option("--vanilla-worker", is_flag=True, default=False)
+@click.option("--node-to-rank-str", type=str, default="")
+@click.option("--master-addr", type=str, default="")
+@click.option("--master-port", type=int, default=0)
+def main(
+    num_workers: int = 4,
+    num_epochs: int = 4,
+    use_gpu: bool = False,
+    vanilla_worker: bool = False,
+    node_to_rank_str: str = "",
+    master_addr: str = "",
+    master_port: int = 0,
+):
+    config = {"lr": 1e-3, "batch_size": 64, "epochs": num_epochs}
+
+    if vanilla_worker:
+        # This path is invoked on every worker when kicking off vanilla training.
+        # First we parse the node to rank string into a map and then find out
+        # our local rank.
+        node_ip_to_ranks = {
+            ip: rank
+            for ip, rank in [
+                ip_rank.split(":", maxsplit=1)
+                for ip_rank in node_to_rank_str.split(",")
+            ]
+        }
+        node_ip = socket.gethostbyname(socket.gethostname())
+        rank = int(node_ip_to_ranks[node_ip])
+
+        # Then we kick off the training function on every worker.
+        return train_torch_vanilla_worker(
+            config=config,
+            rank=rank,
+            world_size=num_workers,
+            master_addr=master_addr,
+            master_port=master_port,
+        )
+
+    import ray
+    from benchmark_util import upload_file_to_all_nodes, run_command_on_all_nodes
+
+    ray.init("auto")
+    print("Preparing Torch benchmark: Downloading MNIST")
+
+    path = os.path.abspath("workloads/_torch_prepare.py")
+    upload_file_to_all_nodes(path)
+    run_command_on_all_nodes(["python", path])
+
+    print("Running Torch Ray benchmark")
+
+    start_time = time.monotonic()
+    train_torch_ray_air(num_workers=4, use_gpu=use_gpu, config=config)
+    time_taken = time.monotonic() - start_time
+
+    time_ray = time_taken
+
+    print(f"Finished Ray training ({num_epochs} epochs) in {time_taken:.2f} seconds.")
+
+    print("Running Torch vanilla benchmark")
+
+    start_time = time.monotonic()
+    train_torch_vanilla(num_workers=4, use_gpu=use_gpu, config=config)
+    time_taken = time.monotonic() - start_time
+
+    time_vanilla = time_taken
+
+    print(
+        f"Finished vanilla training ({num_epochs} epochs) in {time_taken:.2f} seconds."
+    )
+
+    result = {
+        "torch_mnist_ray": time_ray,
+        "torch_mnist_vanilla": time_vanilla,
+    }
+    test_output_json = os.environ.get("TEST_OUTPUT_JSON", "/tmp/result.json")
```

Comment:
Do we have to explicitly fail here if the times are not within some threshold of each other?

Reply:
If we track these times on a dashboard we should be fine, but yeah, we could introduce thresholds if needed, e.g. 95% or so.

Reply:
(obviously if we don't fulfill that criteria in all benchmarks we can adjust it)

Reply:
Added with `time_ray / time_vanilla > 1.15`